source/SpinalHDL/Libraries/fiber.rst

```diff
@@ -10,22 +10,23 @@ Currently in developpement.
 
 The Fiber to run the hardware elaboration in a out of order manner, a bit similarly to Makefile, where you can define rules and dependencies which will then be solved when you run a make command. It is very similar to the Scala Future feature.
 
-Such framework complexify simple things but provide some strong feature for complex cases :
+Using this framework can complicate simple things but provide some strong features for complex cases :
 
-- You can define things before even knowing all their requirements, ex : instanciating a interruption controller, before knowing how many lines of interrupt you need
+- You can define things before even knowing all their requirements, ex :
+  instantiating a interruption controller, before knowing how many interrupt signal lines you need
 - Abstract/lazy/partial SoC architecture definition allowing the creation of SoC template for further specialisations
-- Automatic requirements negotiation between multiple agents in a decentralized way, ex : between masters and slaves of a memory bus
+- Automatic requirement negotiation between multiple agents in a decentralized way, ex : between masters and slaves of a memory bus
 
 The framework is mainly composed of : 
 
 - Handle[T], which can be used later to store a value of type T. 
 - handle.load which allow to set the value of a handle (will reschedule all tasks waiting on it)
 - handle.get, which return the value of the given handle. Will block the task execution if that handle isn't loaded yet
 - Handle{ code }, which fork a new task which will execute the given code. The result of that code will be loaded into the Handle
-- soon(handle), which allow the current task to announce that soon it will load that handle with a value (used to track which handle will
+- soon(handle), which allow the current task to announce that soon it will load that handle with a value (used to track which handle will ???REVIEWME)
```

Comment:
Note to self, still need to provide this.

Reply:
I do not know the fiber system well enough to comment.

This line of the original description did not help to understand the purpose of soon(handle).

Your new description explains the purpose of part of the API, so this is better :)